invenio_access/permissions.py

```diff
@@ -117,64 +118,78 @@ def _cache_key(action_need):
         """Helper method to generate cache key."""
         return get_action_cache_key(
             action_need.value,
-            action_need.argument if hasattr(action_need, 'argument') else None
+            action_need.argument if hasattr(action_need, "argument") else None,
         )
 
+    @staticmethod
+    def _split_actionsneeds(needs):
+        """Split needs into sets of ActionNeed and any other *Need."""
+        action_needs, other_needs = set(), set()
+        for need in needs:
+            action_needs.add(need) if need.method == "action" else \
```

Comment:
Minor: Giving the fact that the if/else is in one line. Wouldn't a lambda be better?

Reply:
Changed as discussed